### PR TITLE
Update issue templates to use forms and improve all templates

### LINKED ISSUE #1747: Convert issue and PR templates to use forms
**Description**
Sometimes users do not follow the provided template when opening issue or PRs. GitHub [supports the use of forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms) for these, so existing templates should be updated to use forms.

**Requirements**
Issues and PR templates that use forms instead of pre-formatted normal text areas.

**Acceptance Criteria (Definition of Done)**
All issue and PR templates converted to forms.

**Dependencies**
None

### PR DESCRIPTION
**Pull request by WalterKolczynski-NOAA**
# Description
Replaces the markdown-based templates for new issues to form-based templates for a better user experience and more consistent responses. Also improves all of the templates, including the PR template. The PR template was not able to be converted to a form as that is not (yet) supported by GitHub.

Issues all apply the "triage" label pending assignment to a project.

Resolves #1747

# Type of change
- Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change requires a documentation update? NO

# How has this been tested?
Examined forms in GitHub code preview.

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/NCO_bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  - type: textarea
+    id: description
+    attributes:
+      label: Additional information
+      description: Provide context or any additional information about the bug.
+    validations:
+      required: true
+
+  - type: textarea
+    id: propsed_implementation
```
Comment (KateFriedman-NOAA) [inline .github/ISSUE_TEMPLATE/NCO_bug_report.yml]: Missing an "o" in the id.

**Review by KateFriedman-NOAA — APPROVED** [review]
Love this! Only one tiny comment left.

Is it possible to view these forms before submitting this?

**Comment by WalterKolczynski-NOAA** [comment]
@KateFriedman-NOAA if you go to the branch with the changes and view the file, it _kinda_ shows what it will look like.

File: .github/ISSUE_TEMPLATE/NCO_bug_report.yml (hunk 1 above)
Comment (WalterKolczynski-NOAA) [reply .github/ISSUE_TEMPLATE/NCO_bug_report.yml]: Fixed.

**Comment by KateFriedman-NOAA** [comment]
> @KateFriedman-NOAA if you go to the branch with the changes and view the file, it _kinda_ shows what it will look like.

Ah! Nice. Yep, I can get the gist of it there. Thanks!

**Review by aerorahul — APPROVED** [review]
Looks good.

**Pull request merged**